File: ios/Shaker/Purchasely/SubscriptionInfo.swift
import Foundation

/// App-level view of an active Purchasely subscription. SDK-free on purpose:
/// `PurchaselyWrapper.fetchSubscriptions` maps the SDK's `PLYSubscription` to
/// this type so the data layer never imports `Purchasely`.
/// Mirrors the Android `SubscriptionInfo`.
struct SubscriptionInfo {
    let planName: String?
    let productName: String?
    let isActive: Bool
}

File: android/app/src/main/java/com/purchasely/shaker/purchasely/SubscriptionInfo.kt
package com.purchasely.shaker.purchasely

/**
 * App-level view of an active Purchasely subscription. SDK-free on purpose:
 * `PurchaselyWrapper.fetchSubscriptions` maps the SDK's `PLYSubscriptionData`
 * to this type so the data layer never imports `io.purchasely`.
 */
data class SubscriptionInfo(
    val planName: String?,
    val productName: String?,
    val isActive: Boolean,
)
